samplers/src/test/java/internal/CelBasedSamplerComponentProviderTest.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package internal;
+
+import static io.opentelemetry.sdk.trace.samplers.SamplingResult.recordAndSample;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import dev.cel.common.CelValidationException;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigException;
+import io.opentelemetry.api.incubator.config.DeclarativeConfigProperties;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.contrib.sampler.CelBasedSampler;
+import io.opentelemetry.contrib.sampler.CelBasedSamplerBuilder;
+import io.opentelemetry.contrib.sampler.internal.CelBasedSamplerComponentProvider;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.DeclarativeConfiguration;
+import io.opentelemetry.sdk.trace.IdGenerator;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import io.opentelemetry.semconv.HttpAttributes;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class CelBasedSamplerComponentProviderTest {
+
+  private static final CelBasedSamplerComponentProvider PROVIDER =
+      new CelBasedSamplerComponentProvider();
+
+  @Test
+  void endToEnd() {
+    String yaml =
+        "file_format: 0.4\n"
+            + "tracer_provider:\n"
+            + "  sampler:\n"
+            + "    parent_based:\n"
+            + "      root:\n"
+            + "        cel_based:\n"
+            + "          fallback_sampler:\n"
+            + "            always_on:\n"
+            + "          expressions:\n"
+            + "            - action: DROP\n"
+            + "              expression: '\"example.com\" in attribute[\"http.response.header.host\"] && attribute[\"http.response.status_code\"] == 200'\n"
+            + "            - action: DROP\n"
+            + "              expression: spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")\n"
+            + "            - action: RECORD_AND_SAMPLE\n"
+            + "              expression: spanKind == \"SERVER\" && attribute[\"url.path\"].matches(\"/actuator.*\")\n";
+    OpenTelemetrySdk openTelemetrySdk =
+        DeclarativeConfiguration.parseAndCreate(
+            new ByteArrayInputStream(yaml.getBytes(StandardCharsets.UTF_8)));
+    Sampler sampler = openTelemetrySdk.getSdkTracerProvider().getSampler();
+    assertThat(sampler.toString())
```

Comment:
I think it's fine for this first pass (shouldn't gate this PR), but testing through the `toString()` is a bit of an antipattern. We can look at ways of making that more testable later tho.

Reply:
I did a total rework of this to replace the `toString()`.